### PR TITLE
[Winlogbeat] Add subdomain value for sysmon module

### PR DESCRIPTION
**Pull request by andrewstucki**
## What does this PR do?

This adds `subdomain` to the event in the sysmon module. To make it a bit more reusable I added an optional `target_subdomain_field` to the `registered_domain` processor that different modules in winlogbeat and filebeat use for parsing out the `registered_domain` value.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Relates https://github.com/elastic/beats/issues/21674

### CODE REVIEW

File: libbeat/processors/registered_domain/docs/registered_domain.asciidoc [hunk 1]
```diff
@@ -9,7 +9,8 @@ The `registered_domain` processor reads a field containing a hostname and then
 writes the "registered domain" contained in the hostname to the target field.
 For example, given `www.google.co.uk` the processor would output `google.co.uk`.
 In other words the "registered domain" is the effective top-level domain
-(`co.uk`) plus one level (`google`).
+(`co.uk`) plus one level (`google`). Optionally, it can store the rest of the
+domain, the `subdomain` into another target field.
 
 This processor uses the Mozilla Public Suffix list to determine the value.
```
Comment (andrewkroh) [inline libbeat/processors/registered_domain/docs/registered_domain.asciidoc:15]: Can you please update the example below with the new config option to encourage use.

**Review by andrewkroh — APPROVED** [review]
Good idea to update the processor.

I noticed there is a `dns.question.top_level_domain` field. I guess. that. could go in there too. https://www.elastic.co/guide/en/ecs/current/ecs-dns.html#field-dns-question-top-level-domain

File: libbeat/processors/registered_domain/docs/registered_domain.asciidoc (hunk 1 above)
Comment (andrewstucki) [reply libbeat/processors/registered_domain/docs/registered_domain.asciidoc:15]: Sounds good.

For the `top_level_domain` I'll just go ahead and create an issue for it. If I have time to add it to the processor after the rest of the ECS upgrade I'll go ahead and tackle it.

**Comment by andrewstucki** [comment]
Opened an issue to track circling back and adding "top_level_domain" handling: https://github.com/elastic/beats/issues/23005

**Pull request merged**